pennylane/devices/default_tensor.py

```diff
@@ -337,24 +313,39 @@ def dtype(self):
         """Tensor complex data type."""
         return self._dtype
 
-    def _reset_state(self) -> None:
+    def _reset_mps(self, wires: qml.wires.Wires) -> None:
         """
-        Reset the MPS.
+        Reset the MPS associated with the circuit.
+
+        Internally, it uses `quimb`'s `CircuitMPS` class.
 
-        This method modifies the tensor state of the device.
+        Args:
+            wires (Wires): The wires to reset the MPS.
         """
-        self._circuitMPS = qtn.CircuitMPS(psi0=self._initial_mps())
+        self._quimb_mps = qtn.CircuitMPS(
+            psi0=self._initial_mps(wires),
+            max_bond=self._max_bond_dim,
+            gate_contract=self._contract,
+            cutoff=self._cutoff,
+        )
 
-    def _initial_mps(self) -> "qtn.MatrixProductState":
+    def _initial_mps(self, wires: qml.wires.Wires) -> "qtn.MatrixProductState":
         r"""
         Return an initial state to :math:`\ket{0}`.
 
         Internally, it uses `quimb`'s `MPS_computational_state` method.
 
+        Args:
+            wires (Wires): The wires to initialize the MPS.
+
         Returns:
             MatrixProductState: The initial MPS of a circuit.
         """
-        return qtn.MPS_computational_state(**self._init_state_opts)
+        return qtn.MPS_computational_state(
```

Comment:
Could we just set `zero state` with previous state?

Reply:
I am not sure I understand. We can discuss this later via Google :+1: